### ISSUE TITLE
InputDecoder NoSuchMethodError

### ISSUE THREAD

**Issue by harmanea** [nom-tam-fits/nom-tam-fits#252]
Hello, I've just updated the library to the new 1.16.0 release from the 1.15.3-SNAPSHOT and I'm getting the following error when reading data from an `ImageHDU`:

```
Exception in thread "main" java.lang.NoSuchMethodError: java.nio.ByteBuffer.rewind()Ljava/nio/ByteBuffer;
	at nom.tam.util.InputDecoder$InputBuffer.loadBytes(InputDecoder.java:321)
	at nom.tam.util.FitsDecoder.read(FitsDecoder.java:554)
	at nom.tam.util.FitsDecoder.readArray(FitsDecoder.java:628)
	at nom.tam.util.FitsDecoder.readArray(FitsDecoder.java:648)
	at nom.tam.util.FitsDecoder.readArray(FitsDecoder.java:648)
	at nom.tam.util.InputDecoder.readArrayFully(InputDecoder.java:190)
	at nom.tam.util.ArrayDataFile.readArrayFully(ArrayDataFile.java:164)
	at nom.tam.image.StandardImageTiler.getCompleteImage(StandardImageTiler.java:305)
	at nom.tam.fits.ImageData.getData(ImageData.java:169)
	at nom.tam.fits.Data.getKernel(Data.java:114)
	at nom.tam.fits.BasicHDU.getKernel(BasicHDU.java:403)
```

This is on linux with openjdk java 1.8.0. Am I doing something wrong or is this a bug?

**Comment by attipaci** [comment]
Hi Adam,

That is really weird. The error claims `ByteBuffer` has no `rewind()` method. According to the API docs, it clearly should have since Java 1.4. I wonder if something got corrupted in your Java install...

I'll think about it anyway, just in case. And if either you or I can point a finger at something in the library, we can fix that for the next update (1.16.1), which I am planning for around March 15. In the meantime, maybe you can try on a different machine (Java install), or with a different Java version (e.g. JDK11) to see if it goes away. Let me know what you find...

**Comment by harmanea** [comment]
I've done some googling and [this](https://www.morling.dev/blog/bytebuffer-and-the-dreaded-nosuchmethoderror/) looks like it explains the problem. When I look at the `MANIFEST.MF` in the `nom-tam-fits-1.16.0.jar` archive, it says `Build-Jdk-Spec: 11`. Could this be the issue?

**Comment by attipaci** [comment]
Interesting....

For now, you can try building your own jar. Just get the source tarball from 'Releases' and run `mvn clean package` in the directory to which you extracted it. After the build, your home-build package will be under `target/`. That should solve your issue in the immediate term.

I'll also see if I can switch the pom to use `--release` instead of the `--source` and `--target` flags when building as your link suggests. Would you be willing to test that change, when I make it, in time for the next update in March?

**Comment by attipaci** [comment]
OK, one side effect of using the `<release>` tag instead of `<source>` and `<target>` for the compiler, is that you will no longer be able to build your own package with Java 8, since the `--release` option was introduced in Java 9 only. I assume you would be OK with that...

**Comment by harmanea** [comment]
Thanks for the quick reply! I'm willing to test any changes you make and I'm okay with not being able to build the package myself.

**Comment by attipaci** [comment]
Thanks in advance. Towards the the end of the month I'll start putting together a release candidate. When it's ready, I'll ping you to verify that the jar works for you out of the box.

**Comment by harmanea** [comment]
I can confirm that this is now fixed in 1.16.1-SNAPSHOT

**Comment by attipaci** [comment]
Thanks for checking so promptly!
